### DISCUSSION TITLE
Alignment to Guidelines

### DISCUSSION THREAD

**Discussion by FabrizioMoggio** [camaraproject/CallForwardingSignal#113] [General]
For Spring25 some alignment to the guidelines is needed.

To everyone, please add comments to underline some alignment needed.

**Comment by rartych** [comment]
@FabrizioMoggio Please have a look at: 
https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/51413132/Analysis+of+Commonalities+0.5.0-alpha.1+changes#Analysis

The table summarizes main changes.

**Comment by FabrizioMoggio** [comment]
List of changes to be implemented:
- [x] I.1: update Commonalities release number from 0.4.0 to 0.5.0
- [x] I.2: error codes update (issue: https://github.com/camaraproject/CallForwardingSignal/issues/112)
- [x] I.3: Authorization and authentication. Info. description update (issue: https://github.com/camaraproject/CallForwardingSignal/issues/120)
- [x] I.4: Enhancements and Alignments in Errors (issue: https://github.com/camaraproject/CallForwardingSignal/issues/119)
- [x] I.5: phone number from access token and 3Legs update (issue: https://github.com/camaraproject/CallForwardingSignal/issues/122)
- [x] I.6: Align API-Testing with Error codes (issue: https://github.com/camaraproject/CallForwardingSignal/issues/123)
- [x] I.7: Phone Number retrieval: https://github.com/camaraproject/CallForwardingSignal/issues/130
- [x] I.8: Phone Number retrieval and Testing: https://github.com/camaraproject/CallForwardingSignal/issues/131

List of the changes (please add what I missed) to be evaluated:
- [x] E.1: Normalization of error status and code allowed values (ref: https://github.com/camaraproject/Commonalities/pull/316) 
- [x] E.2: phone number from access token and 3Legs update (ref: https://github.com/camaraproject/Commonalities/pull/324)
- [x] E.3: Align API-Testing-Guidelines with new Enhancements and Alignments in Errors (ref: https://github.com/camaraproject/Commonalities/pull/343)
- [x] E.4: Enhancements and Alignments in Errors (ref: https://github.com/camaraproject/Commonalities/pull/329)

**Comment by bigludo7** [comment]
@FabrizioMoggio I will add:
- [ ] Add in E.2 -->  https://github.com/camaraproject/Commonalities/pull/329 
- [ ] E.3: Align API-Testing-Guidelines with new Enhancements and Alignments in Errors - we should few small update with the change of error code.

**Reply by FabrizioMoggio** [reply]
perfect. I suppose that #329 is in addition to E.1? right? Anyway, I will add it as a standalone issue

**Reply by FabrizioMoggio** [reply]
analysis on E.4: to be implemented to be aligned with the new error codes text. -> I.4

**Comment by FabrizioMoggio** [comment]
analysis on E.2: to be implemented to be aligned with the section "Identifying the phone number from the access token" in info.description that has been modified in Commonalities 0.5.0 -> I.5

**Comment by FabrizioMoggio** [comment]
analysis on E.1: it is already included in I.2

**Comment by FabrizioMoggio** [comment]
Analysis of E.3: Testing files must be updated because of the update of the error codes (e.g. 409). To be implemented ->I.6

**Comment by FabrizioMoggio** [comment]
Implemented
